lib/Macros/Sources/ObservationMacros/ObservableMacro.swift

```diff
@@ -0,0 +1,159 @@
+import SwiftSyntax
+import SwiftSyntaxMacros
+
+@_implementationOnly import SwiftDiagnostics
+@_implementationOnly import SwiftOperators
+@_implementationOnly import SwiftSyntaxBuilder
+
+private extension DeclSyntaxProtocol {
+  var isObservableStoredProperty: Bool {
+    guard let property = self.as(VariableDeclSyntax.self),
+          let binding = property.bindings.first
+    else {
+      return false
+    }
+
+    return binding.accessor == nil
+  }
+}
+
+public struct ObservableMacro: MemberMacro, MemberAttributeMacro, ConformanceMacro {
+  // MARK: - ConformanceMacro
+  public static func expansion<
+    Declaration: DeclGroupSyntax,
+    Context: MacroExpansionContext
+  >(
+    of node: AttributeSyntax,
+    providingConformancesOf declaration: Declaration,
+    in context: Context
+  ) throws -> [(TypeSyntax, GenericWhereClauseSyntax?)] {
+    let protocolName: TypeSyntax = "Observable"
+    return [(protocolName, nil)]
+  }
+
+  // MARK: - MemberMacro
+  public static func expansion<
+    Declaration: DeclGroupSyntax,
+    Context: MacroExpansionContext
+  >(
+    of node: AttributeSyntax,
+    providingMembersOf declaration: Declaration,
+    in context: Context
+  ) throws -> [DeclSyntax] {
+    guard let identified = declaration.asProtocol(IdentifiedDeclSyntax.self) else {
+      return []
+    }
+
+    let parentName = identified.identifier
+
+    let registrar: DeclSyntax = 
+      """
+      let _registrar = ObservationRegistrar<\(parentName)>()
```

Comment:
The registrar cannot be private or file private since they may have computed properties in extensions or extensions in other files. I guess it could be explicitly `internal` if we wanted.